test/util/cluster/cluster.go

```diff
@@ -105,23 +113,64 @@ func SriovStable(operatorNamespace string, clients *testclient.ClientSet) (bool,
 		return false, nil
 	}
 	for _, state := range nodeStates.Items {
-		if !stateStable(state) {
+		nodeReady, err := stateStable(state, clients, operatorNamespace)
+		if err != nil {
+			return false, err
+		}
+		if !nodeReady {
 			return false, nil
 		}
 	}
 	return true, nil
 }
 
-func stateStable(state sriovv1.SriovNetworkNodeState) bool {
+func stateStable(state sriovv1.SriovNetworkNodeState, clients *testclient.ClientSet, operatorNamespace string) (bool, error) {
 	switch state.Status.SyncStatus {
 	case "Succeeded":
-		return true
+		return CheckReadyGeneration(clients, operatorNamespace, state)
 	// When the config daemon is restarted the status will be empty
 	// This doesn't mean the config was applied
 	case "":
-		return false
+		return false, nil
 	}
-	return false
+	return false, nil
+}
+
+func CheckReadyGeneration(clients *testclient.ClientSet, operatorNamespace string, state sriovv1.SriovNetworkNodeState) (bool, error) {
+	podList, err := clients.Pods(operatorNamespace).List(metav1.ListOptions{LabelSelector: "app=sriov-network-config-daemon"})
+	if err != nil {
+		return false, err
+	}
+
+	var podObj *corev1.Pod
+	for _, pod := range podList.Items {
+		if pod.Spec.NodeName == state.Name {
+			podObj = &pod
+			break
+		}
+	}
+
+	if podObj == nil {
+		return false, fmt.Errorf("failed to find config daemon on the node %s", state.Name)
+	}
+
+	logs, err := pods.GetLog(clients, podObj)
+	if err != nil {
+		return false, err
+	}
+
+	logsList := strings.Split(logs, "\n")
+	for idx, log := range logsList {
+		// example output from the config-daemon
+		// I0412 09:46:26.041882 3910208 writer.go:111] setNodeStateStatus(): syncStatus: Succeeded, lastSyncError:
+		// I0412 09:46:35.293994 3910208 daemon.go:244] nodeStateChangeHandler(): current generation is 183
+		if strings.Contains(log, fmt.Sprintf("current generation is %d", state.Generation)) &&
```

Comment:
There was a patch proposed that may change the output string: https://github.com/openshift/sriov-network-operator/pull/175/files#diff-06b532a2a5a5d3ad1d5341b3e00945efR273